crypto/crypto_test.go

```diff
@@ -258,8 +258,8 @@ func TestValidateSignatureValues(t *testing.T) {
 	check(false, 1, zero, one)
 	check(false, 1, one, zero)
 
-	// correct sig with max r,s
-	check(true, 0, secp256k1nMinus1, secp256k1nMinus1)
+	// correct sig with max r and s above upper limit
+	check(false, 0, secp256k1nMinus1, secp256k1nMinus1)
```

Comment:
Seems related to enforcing that signatures use a "low s-value" and failing if the high version is used.
>  in the context of ECDSA's malleability property, for any given signature (r,s)(r,s) of a message, there exists another valid signature (r,−smod  n)(r,−smodn) for the same message using the same private key. The value −smod  n−smodn would be the complement of ss in the range of possible ss values. So, for every ECDSA signature with a particular ss value, there's a corresponding malleable signature with the complement ss value.

Is there anything else you want to add to this explanation?

Reply:
Yes, exactly. Originally verifySignatureValues method had a homestead bool variable and the test was settled to test with this variable as false, testing a case previous than homestead update. Since we removed this variable to always consider homestead as true the test was failing. Thus I made this change.

https://github.com/dominant-strategies/go-quai/blob/main/crypto/crypto.go#L270C16-L270C16

Reply:
See my comment on the other PR, but I think this is the wrong change. This should be a passing signature, otherwise it would be grouped in with the failure cases below it. See the equivalent tests on go-ethereum: https://github.com/ethereum/go-ethereum/blob/25733a4aadba3b60a9766f1e6ac9c787588ba678/crypto/crypto_test.go#L259

Reply:
The reason this test pass in go-ethereum is because on their crypto/crypto.go at ValidateSignatureValues function we have the homestead bool flag as we can see here:
https://github.com/ethereum/go-ethereum/blob/25733a4aadba3b60a9766f1e6ac9c787588ba678/crypto/crypto.go#L262

and at the tests they set the flag as false

https://github.com/ethereum/go-ethereum/blob/25733a4aadba3b60a9766f1e6ac9c787588ba678/crypto/crypto_test.go#L228C1-L229C1


But here at go-quai the flag has been removed, to always be considered true (past homestead fork) as we can see here:
https://github.com/dominant-strategies/go-quai/blob/main/crypto/crypto.go#L264

And at ValidateSignatureValues function we reject upper band S values which is the case of the S value used on this test.

Reply:
Please add a description of this to the commit message, for later reference. Otherwise, I think this PR is ready to go